File: 프로그래머스/src/모음사전.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
// https://school.programmers.co.kr/learn/courses/30/lessons/84512
using namespace std;

string str = "AEIOU";
int countComb = 0;
int answer = 0;
// 중복 조합을 생성하기 위한 DFS 함수
void dfs(string current, int depth, const string& word) {
    if (current == word) {
        answer = countComb;
        return;
    }
    if (depth == 5) {
        return;
    }

    for (int i = 0; i < str.size(); i++) {
        countComb ++;
        string checkWord = current+str[i];
        dfs(current + str[i], depth + 1, word);
    }
}

int solution(string word) {
    dfs("", 0 , word);
    return answer;
}

File: 2026/완전탐색/모음사전.py
# https://school.programmers.co.kr/learn/courses/30/lessons/84512?language=python3
from itertools import *
# 중복 순열 : product(repeat)
def solution(word):
    letters = ['A', 'E', 'I', 'O', 'U']
    def lookup_table():
        tab = []
        for r in range (1,6):
            for x in product(letters, repeat=r):
                tab.append("".join(x))
                # if("".join(x) == word):
                #     return cnt
    # answer = lookup_table();
        return sorted(tab)
    
    def find_word(lookup):
        for x in range(0,len(lookup)):
            if lookup[x] == word:
                return x
    # print(type(lookup_table()))
    answer = find_word(lookup_table()) +1
    return answer